grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java

```diff
@@ -90,38 +94,40 @@ public GrpcClient client() {
 
     @Override
     public EntityType putEntityType(Label label) {
-        return putSchemaConcept(label, ENTITY);
+        return putSchemaConcept(label, ENTITY, () -> getEntityType(label.getValue()));
     }
 
     @Override
     public <V> AttributeType<V> putAttributeType(Label label, AttributeType.DataType<V> dataType) {
-        return putSchemaConcept(label, ATTRIBUTE, var -> var.datatype(dataType));
+        return putSchemaConcept(label, ATTRIBUTE, () -> getAttributeType(label.getValue()), var -> var.datatype(dataType));
     }
 
     @Override
     public Rule putRule(Label label, Pattern when, Pattern then) {
-        return putSchemaConcept(label, RULE, var -> var.when(when).then(then));
+        return putSchemaConcept(label, RULE, () -> getRule(label.getValue()), var -> var.when(when).then(then));
     }
 
     @Override
     public RelationshipType putRelationshipType(Label label) {
-        return putSchemaConcept(label, RELATIONSHIP);
+        return putSchemaConcept(label, RELATIONSHIP, () -> getRelationshipType(label.getValue()));
     }
 
     @Override
     public Role putRole(Label label) {
-        return putSchemaConcept(label, ROLE);
+        return putSchemaConcept(label, ROLE, () -> getRole(label.getValue()));
     }
 
-    private <X extends Concept> X putSchemaConcept(Label label, Schema.MetaSchema meta){
-        return putSchemaConcept(label, meta, null);
+    private <X extends SchemaConcept> X putSchemaConcept(Label label, Schema.MetaSchema meta, Supplier<X> supplier){
+        return putSchemaConcept(label, meta, supplier,null);
     }
 
-    private <X extends Concept> X putSchemaConcept(Label label, Schema.MetaSchema meta, @Nullable Function<VarPattern, VarPattern> extender){
+    private <X extends SchemaConcept> X putSchemaConcept(Label label, Schema.MetaSchema meta,
+                                                   Supplier<X> supplier,
+                                                   @Nullable Function<VarPattern, VarPattern> extender){
         VarPattern var = var().label(label).sub(var().label(meta.getLabel()));
         if(extender != null) var = extender.apply(var);
         queryRunner().run(Graql.define(var));
-        return null;
+        return supplier.get();
```

Comment:
Haha, I didn't even notice this was returning `null` before!

Is it possible to just return the answer from the `define` query?

```
return queryRunner.run(Graql.define(var)).get("x").asSchemaConcept();
```